bin/problem.py

```diff
@@ -58,19 +59,31 @@ def _determine_statement_languages(self):
         """
         if isinstance(self.settings.name, str):
             self.settings.name = {'en': self.settings.name}
-        yamlnames = set(self.settings.name)
-        texfiles = set(
+        yamllangs = set(self.settings.name)
+        texlangs = set(
             path.suffixes[0][1:] for path in glob(self.path, 'problem_statement/problem.*.tex')
         )
-        for lang in texfiles - yamlnames:
+        for lang in texlangs - yamllangs:
             error(
                 f"{self.name}: Found problem.{lang}.tex, but no corresponding name in problem.yaml."
             )
-        for lang in yamlnames - texfiles:
+        for lang in yamllangs - texlangs:
             error(
                 f"{self.name}: Found name for language {lang} in problem.yaml, but not problem.{lang}.tex."
             )
-        return sorted(texfiles & yamlnames)
+        for lang in texlangs & yamllangs:
+            unnormalised_yamlname = self.settings.name[lang]
+            yamlname = ' '.join(unnormalised_yamlname.split())
+            with open(self.path / 'problem_statement' / f'problem.{lang}.tex') as texfile:
+                match tex_problem_name := latex.get_argument_for_command(texfile, 'problemname'):
+                    case None:
+                        warn(rf"No \problemname found in problem.{lang}.tex")
+                    case s if s not in ["\\problemyamlname", yamlname]:
```

Comment:
As discussed in the issue, I think it's more reasonable to ensure that the yamlname is a substring of the texname.

The typical reason they differ is because of added typesetting, which shouldn't appear in the yamlname.

Reply:
Solved differently, see below